app/src/main/java/org/schabi/newpipe/database/stream/model/StreamStateEntity.java

```diff
@@ -65,8 +65,8 @@ public void setProgressTime(final long progressTime) {
 
     public boolean isValid(final int durationInSeconds) {
         final int seconds = (int) TimeUnit.MILLISECONDS.toSeconds(progressTime);
-        return seconds > PLAYBACK_SAVE_THRESHOLD_START_SECONDS
-                && seconds < durationInSeconds - PLAYBACK_SAVE_THRESHOLD_END_SECONDS;
+        return (Math.abs(seconds - durationInSeconds) <= 1) || (seconds > PLAYBACK_SAVE_THRESHOLD_START_SECONDS
```

Comment:
It's done like this since for some reason, the duration of a video appears to be rounded up from the actual duration (e.g. for a completed video with a length of 50 seconds, `seconds` might be 49).